### PR TITLE
re-issie - kick banned IP's from the server

### PR DESCRIPTION
**Pull request by kyle95wm**
This will kick players from the server if their IP is in the bans table. This code was tested and confirmed to work on BeanJrFi (my test server)

### CODE REVIEW

**Comment by sepalani** [comment]
As I said in the old PR:

> Do not format the string directly. Use the logger's parameters for that, it does exactly the same without using a tuple with old-style formatting sprintf syntax.

In other word:

``` python
self.log(logging.DEBUG, "**** Banned user, closing network socket for %s|%s...", self.gameid, self.address.host)
```

Not PEP8 compliant. I'm not a huge fan of passing a constructed on the fly dictionary as parameter. The function could either take the appropriate arguments as parameters or use **kwargs. I would opt for the first one since the second one doesn't imply the right key-values are stored in the dict.

By the way, this kind of banning is pretty much inefficient because it's based on the IP address. 

Otherwise, it sounds good to me.

**Comment by kyle95wm** [comment]
Alright I'll make the suggested changes and test them out. I know this ban system is rather basic, but if you could come up with something better or more efficient then sure I'd love to see what you got. 

The whole idea of the above code is to kick the user some time after you've banned them, as from what I'm told there's no way to create a "kick" button on the admin page. Maybe my sources are wrong about this.

**Comment by sepalani** [comment]
We have many pieces of information we can use to ban someone:
- Host IP address
- Host MAC address
- Console MAC address
- Console Friend Code
- User ID/Unique nick
- Profile Friend Code
- Etc.

Banning the IP address doesn't make so much sense, there are, for most people dynamic and will change. Using a proxy will bypass the ban as well. Banning the profile will be more effective and there are plethora way of doing it.

On one hand, you could ban console specific traits making the console unable to play online. On the other hand, you could ban based on the profile data making the profile unreachable for a specific game.

Making a button is possible but assume there is another way to interact with the server than using the Wii. If a such way exists, it mustn't be publicly accessible or it will become a nightmare.

**Comment by kyle95wm** [comment]
does the above meet with your approval? @sepalani 

About a ban system, there was an issue opened long ago about creating one. @SMTDDR and I both created one. In fact if you look at my (now outdated) repo you'll see I did create a console ban system but it was....how shall I put it, messy to say the least. SMTDDR was the one who made the admin page from scratch and I just kind of added onto it

**Comment by sepalani** [comment]
Almost :vP.

According to PEP8:
- Some lines are too long (> 80 characters)
- Missing space after comma
- Missing space after colon

Here is one way to make it PEP8 compliant:

``` python
if self.gameid and self.db.is_banned({'gamecd': self.gameid,
                                      'ipaddr': self.address.host}):
    self.log(logging.DEBUG,
             "**** Banned user, closing network socket for %s|%s...",
             self.gameid, self.address.host)
    self.transport.abortConnection()
    return
```

You can read the [PEP8 norm](https://www.python.org/dev/peps/pep-0008/) if you have any doubt. There is also a tool that can check it automatically for you [here](https://pypi.python.org/pypi/pep8).

In my opinion, you should also squash your commits into a single one, see [this tutorial](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html). Then you'll have to `push --force`.

Anyway, LGTM.

**Comment by kyle95wm** [comment]
so I just copy the above?

**Comment by sepalani** [comment]
Indentations are missing. I advise you not to use github (prefer an IDE, a shell, etc.) when those kind of modification impacts the code. You'll also need to rebase, some materials : [tutorial 1](https://www.youtube.com/watch?v=SxzjZtJwOgo) and [tutorial 2](https://www.youtube.com/watch?v=W39CfI3-JFc). If you aren't familiar with git, learn [basics first](https://try.github.io/levels/1/challenges/1).

**Comment by kyle95wm** [comment]
hmm I seem to have run into a road block with git. I modified the file and then added it. When I went to commit the changes git said that my branch was at the same spot that origin/patch-3 was in. Im confused by this. I just cloned the git repo just now. If you want I can give you write access to my repo

**Comment by sepalani** [comment]
No, thanks. Live & Learn. It's better for you to learn how to use git properly ;v).

Honestly, you should really take the time to watch those videos and practice some basics. You'll save so much time.

If I'm not wrong:
1. First, checkout to the appropriate branch, `git branch --all` to list all branches. According to github it should be `git checkout patch-3`.
2. You should rewrite the file to add the missing indentation
3. Then, `git add` the file.
4. Here you can either `git commit -m "Your commit message"` or `git commit --amend`
5. Right now, you need to do `git rebase -i HEAD~10`, it will rebase the last 10 commits. You'll have to squash **YOUR** commits into a single one.
6. Finally, `git push --force origin patch-3`

**Comment by kyle95wm** [comment]
what about now? Please tell me this is fine or Im going to scream :p

**Comment by sepalani** [comment]
No, indents are still missing. By the way, we use space indentations not tabs, if it can help you. You should really preview your file into an editor to see if there are missing indents or not. And run the server to see if it works normally, of course.

**Comment by kyle95wm** [comment]
I am using spacebar "indents" in fact I direct copy/pasted your code above so it lines up exactly as you have it. Im using a terminal btw now so idk why the indents didn't carry over. 

Also I know how to use git to an extent, for example I reset the head of patch-3 back to the first commit of this PR and amended the commit with the indented changes.

**Comment by sepalani** [comment]
You can't just copy-paste my example above because the indent level isn't the same as the one where the code is located. See the whole file, the indentation level inside your if block is wrong.

**Comment by kyle95wm** [comment]
the "ip" part below the if line, can that stay where it is? (in your example). Im currently trying to work out this whole indentation thing right now

**Comment by kyle95wm** [comment]
I just took another shot. Am I getting closer?

**Comment by sepalani** [comment]
LGTM after a squash of those commits into a single one.

**Comment by kyle95wm** [comment]
so the indenting is correct now?

**Comment by sepalani** [comment]
Ah, there is a missing indent after the string in the log.

File: gamespy_profile_server.py [hunk 1]
```diff
@@ -204,7 +204,13 @@ def rawDataReceived(self, data):
 
             commands, self.remaining_message = \
                 gs_query.parse_gamespy_message(data)
-
+            if self.gameid and self.db.is_banned({'gamecd': self.gameid,
+                                                  'ipaddr': self.address.host}):
+                self.log(logging.DEBUG,
+                    "**** Banned user, closing network socket for %s|%s...",
+                self.gameid, self.address.host)
```
Comment (sepalani) [inline gamespy_profile_server.py]: The indent of:

``` python
self.gameid, self.address.host)
```

must match the string indent to be PEP8 compliant.

EDIT: BTW, the string indent must match the `logging.DEBUG` indent the same way.

**Comment by kyle95wm** [comment]
so line up "self.gameid, self.address.host)" with the open quotes on 

"***\* Banned user, closing network socket for %s|%s...",

**Comment by sepalani** [comment]
In fact, it must look like the example I shown above. The only difference is the indentation level.

File: gamespy_profile_server.py [hunk 2]
```diff
@@ -204,7 +204,13 @@ def rawDataReceived(self, data):
 
             commands, self.remaining_message = \
                 gs_query.parse_gamespy_message(data)
-
+            if self.gameid and self.db.is_banned({'gamecd': self.gameid,
+                                                  'ipaddr': self.address.host}):
+                self.log(logging.DEBUG,
+                    "**** Banned user, closing network socket for %s|%s...",
```
Comment (sepalani) [inline gamespy_profile_server.py]: Must match `logging.DEBUG` indent.

**Comment by kyle95wm** [comment]
alright your example above confuses me for some reason. Do I have the correct indentation or not on those lines?

**Comment by kyle95wm** [comment]
so that means bring self.gameid, self.address.host) back 4 spaces? Still makes no sense

**Comment by sepalani** [comment]
It should look like this:

``` python
self.log(logging.DEBUG,
         "**** Banned user, closing network socket for %s|%s...",
         self.gameid, self.address.host)
```

The first parameter of each line, should be aligned with the first one of the previous line.

**Comment by kyle95wm** [comment]
I hope I got it right this time....

**Comment by sepalani** [comment]
LGTM.

**Comment by kyle95wm** [comment]
alright then

**Comment by kyle95wm** [comment]
Updates?

**Comment by kyle95wm** [comment]
This isn't going anywhere. Might as well close this.

**Pull request closed without merging**